R/data-gather-clean.R

```diff
@@ -0,0 +1,142 @@
+#' Get the submissions based on status
+#'
+#' Get the submissions based on status.
+#'
+#' @inheritParams mod_review_section_server
+#' @param statuses A character vector of statuses to
+#'   include from the set: "Accepted", "Rejected",
+#'   "In Review".
+get_submissions <- function(syn, group, statuses,
+                            section_lookup_table, variable_lookup_table) {
+  if (is.null(statuses)) {
+    return(NULL)
+  }
+  statuses <- purrr::map(
+    statuses,
+    function(x) {
+      switch(
+        x,
+        Accepted = "ACCEPTED",
+        Rejected = "REJECTED",
+        `In Review` = "SUBMITTED_WAITING_FOR_REVIEW"
+      )
+    }
+  )
+  submissions <- synapseforms::download_all_and_get_table(
+    syn,
+    state_filter = statuses[[1]], group = group
+  )
+  if (length(statuses) > 1) {
+    for (status in 2:length(statuses)) {
+      temp_subs <- synapseforms::download_all_and_get_table(
+        syn,
+        state_filter = statuses[[status]], group = group
+      )
+      if (is.null(submissions)) {
+        submissions <- temp_subs
+      } else if (!is.null(temp_subs)) {
+        submissions <- dplyr::full_join(
+          submissions,
+          temp_subs,
+          by = "variables"
+        )
+      }
+    }
+  }
```

Comment:
It took me a minute to understand this. We're getting the data for one status, then if there are more, getting the data for them and joining it in 
Does it work to rewrite this as 

```r
# Get table for each status
submissions <- purrr::map(statuses, function(x) {
  synapseforms::download_all_and_get_table(
    syn = syn,
    state_filter = x,
    group = 9
  )
}) %>%
  purrr::compact() %>% # remove NULLs
  purrr::reduce(full_join, by = "variables") # full_join non-null data frames
```

^^ this should work the same whether one or multiple statuses are selected.

Reply:
Ooh, that's fancy and so clean. I'll try that out. Thanks!

Reply:
This works, as long as I use dplyr::full_join in the reduce function.

I did realize that this will make for an issue with the view all panel since I was mapping the friendlier versions of the statuses to the less friendly versions (ex: "In Review" versus "SUBMITTED_WAITING_FOR_REVIEW") in this function. I'll just fix the mapping in the other branch, which is a better way to do it in the first place.